src/hermes/commands/init/util/git_info.py

```diff
@@ -0,0 +1,106 @@
+# SPDX-FileCopyrightText: 2024 Forschungszentrum Jülich GmbH
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileContributor: Nitai Heeb
+
+import re
+import os
+import subprocess
+from pathlib import Path
+
+
+default_cwd = ""
+
+
+def get_valid_cwd(cwd="") -> str:
+    """
+    Returns the given cwd if valid or the default_cwd / os.getcwd() when not cwd is given.
+    """
+    # Get default when cwd is empty
+    if cwd == "":
+        if default_cwd:
+            cwd = default_cwd
+        else:
+            cwd = os.getcwd()
+    # Test if valid
+    path = Path(cwd)
+    if not path.exists():
+        raise Exception(f"Path {path} does not exist.")
+    elif not path.is_dir():
+        raise Exception(f"Path {path} is not a directory.")
+    # Return
+    return str(path)
+
+
+def run_git_command(command: str, cwd="") -> str:
+    """
+    Runs any git command using subprocess. Raises Exception when returncode != 0.
+    :param command: The command as string with or without the 'git' main command.
+    :param cwd: Path to target directory ( if it differs from os.getcwd() ).
+    :return: The command's output.
+    """
+    # Validate or get default cwd
+    cwd = get_valid_cwd(cwd)
+    # Get command list from string
+    command_list = [s for s in command.split(" ") if s != ""]
+    if command_list[0] != "git":
+        command_list.insert(0, "git")
+    # Run subprocess
+    result = subprocess.run(command_list, cwd=cwd, capture_output=True, text=True)
```

Comment:
I think it would be a good idea to have a common interface for git commands. Or at least to collect them all in some kind of util file. Because calling git is not trivial; see [here](https://github.com/softwarepub/hermes/pull/217).

But I think we can do this later and keep this here for now.